src/app/ubs/ubs-admin/components/add-violations/add-violations.component.ts

```diff
@@ -170,80 +158,71 @@ export class AddViolationsComponent implements OnInit, OnDestroy {
       });
   }
 
-  filesDropped(files: FileHandle[]): void {
+  loadFiles(files: File[]): void {
+    if (this.images.length + files.length > this.maxNumberOfImgs) {
+      return;
+    }
+
     this.isImageSizeError = false;
     this.isImageTypeError = false;
-    this.checkFileExtension(files);
-    const imageFile = files[0].file;
-    this.transferFile(imageFile);
+
+    for (const file of files) {
+      const validExt = this.validateFileExtension(file);
+      const validSize = this.validateFileSize(file);
+      if (!validExt) {
+        this.isImageTypeError = true;
+        continue;
+      }
+      if (!validSize) {
+        this.isImageSizeError = true;
+        continue;
+      }
+      this.transferFile(file);
+      if (this.editMode) {
+        this.isInitialImageDataChanged = true;
+      }
+    }
```

Comment:
also remove continue:
this.isImageTypeError = !this.validateFileExtension(file);
this.isImageSizeError = !this.validateFileSize(file);

if (this.isImageTypeError || this.isImageSizeError) {
this.transferFile(file);
this.isInitialImageDataChanged = this.editMode;
}